derp/derphttp/derphttp_client.go

```diff
@@ -185,6 +185,18 @@ func (c *Client) useHTTPS() bool {
 	if c.url != nil && c.url.Scheme == "http" {
 		return false
 	}
+	// Turn on development mode and connect to DERP using plain HTTP.
+	// Questions:
+	// * Should this be a command line option or an envknob?
```

Comment:
I think envknob is a better fit, since this is for development/testing only, and not something that we'd want users to use (both because it's less secure and because we want to be able to change it at any time). It matches the intent of the package's Go doc: https://github.com/tailscale/tailscale/blob/02582083d54f20e63796a518888aad68ace20b52/envknob/envknob.go#L4-L10

Reply:
Thanks, that helps. What do you think about having an additional envknob to say "ignore the DERP map, just connect to the DERP at this hostname?" I think the infra is all there for that. I'd like to take out the "edit the derp map" step if possible.

Reply:
That additional knob sounds good to me, I would have found it useful.

Reply:
I added a TS_DEBUG_USE_DERP_ADDR which fakes up a derp map/region/node and it's really great.